src/utils.rs

```diff
@@ -42,33 +34,146 @@ where
+                    error!("Task panicked: {:?}", e);
+
+                    if cli_batteries::is_shutting_down() {
+                        std::process::abort();
+                    }
+
+                    tokio::time::sleep(backoff_duration).await;
+                }
             }
         }
     })
 }
 
-#[allow(dead_code)]
+#[cfg(not(feature = "oz-provider"))]
 pub fn u256_to_f64(value: U256) -> f64 {
-    value.to_string().parse::<f64>().unwrap()
+    value
+        .to_string()
+        .parse::<f64>()
+        .expect("Failed to parse U256 to f64")
+}
+
+#[cfg(test)]
+mod tests {
+    use std::{
+        sync::{
+            atomic::{AtomicBool, Ordering},
+            Arc,
+        },
+        time::Duration,
+    };
+
+    use super::*;
+
+    #[cfg(not(feature = "oz-provider"))]
+    mod u256_to_f64_tests {
+        #![allow(clippy::float_cmp)]
+
+        use hex_literal::hex;
+        use test_case::test_case;
+
+        use super::*;
+
+        #[test_case(1_000_000_000_000_000_000 => 1_000_000_000_000_000_000.0)]
+        #[test_case(42 => 42.0)]
+        #[test_case(0 => 0.0)]
+        fn test_u256_to_f64_small(v: u64) -> f64 {
+            u256_to_f64(U256::from(v))
+        }
+
+        #[test_case(hex!("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF") => 115_792_089_237_316_195_423_570_985_008_687_907_853_269_984_665_640_564_039_457_584_007_913_129_639_935.0)]
+        #[test_case(hex!("0FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF") => 7_237_005_577_332_262_213_973_186_563_042_994_240_829_374_041_602_535_252_466_099_000_494_570_602_495.0)]
+        fn test_u256_to_f64_large(v: [u8; 32]) -> f64 {
+            u256_to_f64(U256::from(v))
+        }
+    }
+
+    #[tokio::test]
+    async fn spawn_monitored_test() -> anyhow::Result<()> {
+        let can_finish = Arc::new(AtomicBool::new(false));
+        let triggered_error = Arc::new(AtomicBool::new(false));
+
+        let handle = {
+            let can_finish = can_finish.clone();
+            let triggered_error = triggered_error.clone();
+
+            spawn_monitored_with_backoff(
+                move || {
+                    let can_finish = can_finish.clone();
+                    let triggered_error = triggered_error.clone();
+
+                    async move {
+                        let can_finish = can_finish.load(Ordering::SeqCst);
+
+                        if can_finish {
+                            Ok(())
+                        } else {
+                            triggered_error.store(true, Ordering::SeqCst);
+
+                            // Sleep a little to free up the executor
+                            tokio::time::sleep(Duration::from_millis(20)).await;
+
+                            panic!("Panicking!");
+                        }
+                    }
+                },
+                Duration::from_secs_f32(0.2),
+            )
+        };
+
+        println!("Sleeping for 1 second");
```

Comment:
`info!`?

Reply:
Actually this is test output and I'm not spinning up a tracing subscriber, so that wouldn't work. I'll leave the printlns